content/blog/2022/dependabot-single-pull-request.md

```diff
@@ -0,0 +1,220 @@
+2022, the feature is on [Dependabot's
+roadmap](https://github.com/github/roadmap/issues/148), but doesn't have a
+concrete release date.
+
+At OddBird we decided to leverage [GitHub
+Actions](https://github.com/features/actions) to produce a pull request with all
+dependencies updated at once. We have created a workflow that runs every Monday
+in the early morning. By the time we start our work day the pull request has
+been opened and we can merge it right away if all tests pass, or debug any
+issues introduced by upstream breaking changes. We also have the option of
+[running the workflow on
+demand](https://docs.github.com/en/actions/managing-workflow-runs/manually-running-a-workflow)
+to upgrade dependencies at our convenience.
+
+Broadly speaking, the workflow will:
+
+1. Checkout our code
+1. Configure the runtimes required by our project (Python and Node)
+1. Run a custom command to upgrade all dependencies
+1. Use git to detect changed files
+1. Commit the changes to a pre-defined branch (if they exist)
+1. Open a pull request for that branch (if it doesn't exist already)
+
+This workflow should be compatible with most projects, but you will need to
+adjust steps two and three to run your own commands depending on what package
+manager you use. We'll explain how to do it with
+[pip-tools](https://github.com/jazzband/pip-tools) for Python and
+[yarn](https://yarnpkg.com/) for Node.
+
+## Upgrade Python Dependencies
+
+By using GitHub's own [setup-python](https://github.com/actions/setup-python)
+Action we can easily install our required Python version on the runner and
+configure dependency caching. Once the Python runtime has been set up we just
+need to call `pip-tools` to upgrade all dependencies:
+
+```bash
+pip install -U pip pip-tools
+pip-compile --upgrade -o requirements/prod.txt requirements/prod.in
+pip-compile --upgrade -o requirements/dev.txt requirements/dev.in
+```
+
+This results in changes to `.txt` and `.in` files in the requirements folder.
+These changes will be included in the automated pull request and CI will test
+the project against the new package versions.
+
+Note that if you use other tools to install Python packages (like Poetry,
+Pipenv, or plain pip) you will need to use those instead of pip-tools to
+generate the upgraded requirement files.
+
+## Upgrade Node (JavaScript) Dependencies
+
+GitHub also provides an official
+[setup-node](https://github.com/actions/setup-node) Action that will configure
+Node on the runner. After the Node runtime has been set up we can upgrade Node
+dependencies with:
+
+```bash
+rm -f yarn.lock
+npx --yes yarn-upgrade-all
+```
+
+In this case we are using
+[`yarn-upgrade-all`](https://yarnpkg.com/package/yarn-upgrade-all) to get
+updated versions for all dependencies. This will modify both `package.json` and
+`yarn.lock`, which will cause CI to test against the new package versions when
+the automated pull request is opened.
```

Comment:
IMO, one of the main advantages of `yarn-upgrade-all` is that it allows us to specify individual dependencies to [ignore](https://github.com/tylerlong/yarn-upgrade-all#ignore-some-packages) -- I think it'd be worth mentioning that, and maybe even showing a snippet of what that looks like in our `package.json` file?